test/metabase/task/email_remove_legacy_pulse_test.clj

```diff
@@ -0,0 +1,36 @@
+(ns ^:mb/once metabase.task.email-remove-legacy-pulse-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase.task.email-remove-legacy-pulse :as email-remove-legacy-pulse]
+   [metabase.test :as mt]
+   [toucan2.core :as t2]))
+
+(deftest send-emails-to-admins-if-instance-has-legacy-pulse
+  (mt/with-temp
+    [:model/User  admin  {:is_superuser true :first_name "Ngoc" :last_name "Khuat"}
+     :model/User  _user  {:is_superuser false}
+     :model/Pulse _      {:name "Legacy pulse" :dashboard_id nil :alert_condition nil}
+     :model/Pulse _      {:name "Archived pulse" :dashboard_id nil :alert_condition nil :archived true}]
+    (mt/with-fake-inbox
+      (#'email-remove-legacy-pulse/email-remove-legacy-pulse)
+      (testing "all receivers are superuser"
+        (is (every? true? (t2/select-fn-vec :is_superuser  :model/User :email [:in (keys @mt/inbox)]))))
+      (let [found-regexes  [#"Hi Ngoc Khuat,"
+                            #"<li><a href=\"https?://[^\/]+\/pulse/\d+\">Legacy pulse<\/a></li>"]
+            not-found-re   #"<li><a href=\"https?://[^\/]+\/pulse/\d+\">Archived pulse<\/a></li>"
+            expected-email {:subject "Removal of Legacy Pulses in Upcoming Metabase Release"
+                            :body    (merge (zipmap (map str found-regexes) (repeat true))
+                                            {(str not-found-re) false})}]
+        (testing "the email should say Hi and contains link to active pulses"
+          (is (= (mt/email-to admin expected-email)
+                 (select-keys (apply mt/regex-email-bodies (conj found-regexes not-found-re)) [(:email admin)]))))))))
```

Comment:
You should test that non-admins don't get the email

Reply:
yes I checked that on line 17

Reply:
ah, of course